docs/thrift.md

```diff
@@ -0,0 +1,84 @@
+# Zanzibar Thrift file semantics
+
+This document defines the semantics of a thrift file.
+
+# HTTP + JSON
+
+## Structs and types
+
+The structs defined in the thrift file are serialized into
+JSON. When parsing, any fields on the wire not defined
+in the thrift struct are ignored.
+
+## Annotations
+
+### `zanzibar.http.method`
+
+required.
+
+The HTTP method to use, this is mandatory and can be
+"GET", "POST", "DELETE", "UPDATE", "PATCH"
+
+If the method is GET then the function arguments
+must have http.ref annotations remapping arguments
+to query parameters or params.
```

Comment:
This line might not be necessary as the customization is specified in zanzibar.http.ref section.